### PR TITLE
Add Test for reverse method (reverseTest)

### PR DESCRIPTION
**Pull request by ALeksey5678**

### CODE REVIEW

File: src/com/company/reverse/Reverse.java [hunk 1]
```diff
@@ -2,11 +2,16 @@
 
 
 public class Reverse {
-    private static final String MATCH_ALL_DIGITS = "[\\d+]";
+    private static final String MATCH_ALL_EXCEPT_UNICODE_LETTERS = "[^\\p{L}]+";
     private static final String EMPTY = "";
 
-    public String getReversed(String stringToReverse) {
-        StringBuilder reverseString = new StringBuilder(stringToReverse.replaceAll(MATCH_ALL_DIGITS, EMPTY)).reverse();
+    public String getReversed(String stringToReverse) { //как тестировать эксепшены
```
Comment (saber0008) [inline src/com/company/reverse/Reverse.java]: please clean up comments before doing commit

File: src/com/company/reverse/Reverse.java [hunk 2]
```diff
@@ -2,11 +2,16 @@
...
+        if (stringToReverse == null) {
+
```
Comment (saber0008) [inline src/com/company/reverse/Reverse.java]: please remove empty lines

File: src/com/company/reverse/ReverseTest.java [hunk 3]
```diff
@@ -0,0 +1,31 @@
+package com.company.reverse;
```
Comment (saber0008) [inline src/com/company/reverse/ReverseTest.java]: please move tests to specific directory for tests

Comment (saber0008) [reply src/com/company/reverse/ReverseTest.java]: https://maven.apache.org/guides/introduction/introduction-to-the-standard-directory-layout.html

File: src/com/company/reverse/ReverseTest.java [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+class ReverseTest {
+    Reverse reverse = new Reverse();
+
+    @Test
+    void stringShouldReturnReversedStringWithDigitsUnchanged() {
```
Comment (saber0008) [inline src/com/company/reverse/ReverseTest.java]: please remove string in name

File: src/com/company/reverse/ReverseTest.java [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+    }
+
+    @Test
+    void stringShouldReturnReversedStringWithDigitsSpacesSymbolsUnchanged() {
+        assertEquals("!b1 !A2", reverse.getReversed("!A1 !b2"));
+    }
+
+    @Test
+    void getException() {
+
```
Comment (saber0008) [inline src/com/company/reverse/ReverseTest.java]: empty line

File: src/com/company/reverse/ReverseTest.java [hunk 6]
```diff
@@ -0,0 +1,31 @@
+package com.company.reverse;
+
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+class ReverseTest {
+    Reverse reverse = new Reverse();
```
Comment (saber0008) [inline src/com/company/reverse/ReverseTest.java]: please use access modifiers for class fields

File: src/com/company/reverse/ReverseTest.java [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+
+    @Test
+    void stringShouldReturnReversedStringWithDigitsSpacesSymbolsUnchanged() {
+        assertEquals("!b1 !A2", reverse.getReversed("!A1 !b2"));
+    }
+
+    @Test
+    void getException() {
+
+        assertThrows(IllegalArgumentException.class, () -> {reverse.getReversed(null);});
```
Comment (saber0008) [inline src/com/company/reverse/ReverseTest.java]: please remove brackets and semicolon when writing single line lambdas

File: src/com/company/reverse/ReverseTest.java [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+        assertEquals("dcba1", reverse.getReversed("abcd1"));
+    }
+
+    @Test
+    void stringShouldReturnReversedStringWithDigitsSpacesSymbolsUnchanged() {
+        assertEquals("!b1 !A2", reverse.getReversed("!A1 !b2"));
+    }
+
+    @Test
+    void getException() {
```
Comment (saber0008) [inline src/com/company/reverse/ReverseTest.java]: please give it more suitable name

**Review by saber0008 — CHANGES_REQUESTED** [review]

File: src/com/company/reverse/ReverseTest.java [hunk 9]
```diff
@@ -0,0 +1,31 @@
+package com.company.reverse;
+
+import org.junit.jupiter.api.BeforeAll;
```
Comment (saber0008) [inline src/com/company/reverse/ReverseTest.java]: please clean up unused imports

Comment (ALeksey5678) [reply src/com/company/reverse/ReverseTest.java]: Done

File: src/com/company/reverse/ReverseTest.java (hunk 8 above)
Comment (ALeksey5678) [reply src/com/company/reverse/ReverseTest.java]: Done

File: src/com/company/reverse/ReverseTest.java (hunk 7 above)
Comment (ALeksey5678) [reply src/com/company/reverse/ReverseTest.java]: Done

File: src/com/company/reverse/ReverseTest.java (hunk 6 above)
Comment (ALeksey5678) [reply src/com/company/reverse/ReverseTest.java]: Done

File: src/com/company/reverse/ReverseTest.java (hunk 5 above)
Comment (ALeksey5678) [reply src/com/company/reverse/ReverseTest.java]: Done

File: src/com/company/reverse/ReverseTest.java (hunk 4 above)
Comment (ALeksey5678) [reply src/com/company/reverse/ReverseTest.java]: Done

File: src/com/company/reverse/Reverse.java (hunk 1 above)
Comment (ALeksey5678) [reply src/com/company/reverse/Reverse.java]: Done

File: src/com/company/reverse/Reverse.java (hunk 2 above)
Comment (ALeksey5678) [reply src/com/company/reverse/Reverse.java]: Done

File: src/test/ReverseTest.java [hunk 10]
```diff
@@ -0,0 +1,27 @@
+import com.company.reverse.Reverse;
```
Comment (saber0008) [inline src/test/ReverseTest.java]: test classes should be in same package as class under the test

File: src/com/company/reverse/Reverse.java [hunk 11]
```diff
@@ -2,11 +2,14 @@
```
Comment (saber0008) [inline src/com/company/reverse/Reverse.java:2]: please follow default project structure 
```
src/
    main/
      java/
    test/
      java/
```

Comment (saber0008) [reply src/com/company/reverse/Reverse.java:2]: in this case please put main code to `main` directory

File: src/test/ReverseTest.java [hunk 12]
```diff
@@ -0,0 +1,27 @@
...
+    }
+
+    @Test
+    void shouldReplaceDigitsSpacesSymbols() {
+        assertEquals("!b1 !A2", reverse.getReversed("!A1 !b2"));
+    }
+
+    @Test
+    void shouldThrowIllegalArgumentException() {
+        assertThrows(IllegalArgumentException.class,()-> reverse.getReversed(null));
```
Comment (saber0008) [inline src/test/ReverseTest.java]: please put whitespace after comma

File: src/test/ReverseTest.java [hunk 13]
```diff
@@ -0,0 +1,27 @@
...
+        assertEquals("dcba1", reverse.getReversed("abcd1"));
+    }
+
+    @Test
+    void shouldReplaceDigitsSpacesSymbols() {
+        assertEquals("!b1 !A2", reverse.getReversed("!A1 !b2"));
+    }
+
+    @Test
+    void shouldThrowIllegalArgumentException() {
```
Comment (saber0008) [inline src/test/ReverseTest.java]: please add condition under which this exception can be thrown.
In your case we can name it `shouldThrowIllegalArgumentExceptionIfNull`

File: src/test/ReverseTest.java [hunk 14]
```diff
@@ -0,0 +1,27 @@
...
+        assertThrows(IllegalArgumentException.class,()-> reverse.getReversed(null));
+    }
+
+    @Test
+    void shouldNotReplaceDigitsAndSymbols() {
+        assertEquals("!1asd2!", reverse.getReversed("!1dsa2!"));
+    }
+}
```
Comment (saber0008) [inline src/test/ReverseTest.java]: please add additional test for empty string

**Review by saber0008 — CHANGES_REQUESTED** [review]

File: src/test/ReverseTest.java (hunk 10 above)
Comment (ALeksey5678) [reply src/test/ReverseTest.java]: Done

File: src/com/company/reverse/Reverse.java (hunk 11 above)
Comment (ALeksey5678) [reply src/com/company/reverse/Reverse.java:2]: Done

File: src/test/ReverseTest.java (hunk 12 above)
Comment (ALeksey5678) [reply src/test/ReverseTest.java]: Done

File: src/test/ReverseTest.java (hunk 13 above)
Comment (ALeksey5678) [reply src/test/ReverseTest.java]: Done

File: src/test/ReverseTest.java (hunk 14 above)
Comment (ALeksey5678) [reply src/test/ReverseTest.java]: Done

File: src/test/java/com/company/reverse/ReverseTest.java [hunk 15]
```diff
@@ -27,4 +27,7 @@ void shouldNotReplaceDigitsAndSymbols() {
         assertEquals("!1asd2!", reverse.getReversed("!1dsa2!"));
     }
 
+    @Test
+    void shouldReturnEmptySpace() { assertEquals("", reverse.getReversed(""));
```
Comment (saber0008) [inline src/test/java/com/company/reverse/ReverseTest.java]: Please fix formatting

**Review by saber0008 — CHANGES_REQUESTED** [review]
Looks good
Only one minor issue with formatting

**Review by saber0008 — APPROVED** [review]
